weaver/processes/utils.py

```diff
@@ -84,29 +86,42 @@ def _read_reference(url):
         return None
 
 
-def _get_multi_json_references(output):
-    # type: (owslib.wps.Output) -> Optional[List[JSON]]
+def _get_multi_json_references(output, container):
+    # type: (owslib.wps.Output, Optional[AnySettingsContainer]) -> Optional[List[JSON]]
     """
     Since WPS standard does not allow to return multiple values for a single output,
     a lot of process actually return a json array containing references to these outputs.
-    This function goal is to detect this particular format.
 
-    :return: An array of HTTP(S) references if the specified output is effectively a JSON containing that,
-             ``None`` otherwise.
-    """
+    Because the multi-output references are contained within this JSON file, it is not very convenient to retrieve
+    the list of URLs as one always needs to open and read the file to get them. This function goal is to detect this
+    particular format and expand the references to make them quickly available in the job output response.
 
-    # Check for the json datatype and mimetype
+    :return:
+        Array of HTTP(S) references if the specified output is effectively a JSON containing that, ``None`` otherwise.
+    """
+    # Check for the json datatype and mime-type
     if output.dataType == WPS_COMPLEX_DATA and output.mimeType == CONTENT_TYPE_APP_JSON:
+        try:
+            # If the json data is referenced read it's content
+            if output.reference:
+                out_ref = output.reference
+                if container:
+                    if out_ref.startswith("file://"):
+                        out_ref = out_ref[7:]
+                    if out_ref.startswith("/"):
```

Comment:
absolute file format used as relative path? Why not starting with a "."?

Reply:
URL path, not directory path
for convenience, files are saved in db as /job-id.... so that we just add wps-outputs path in front regardless of (possibly changing) setting